Make ReferenceQvm compatible with pyQuil v1.4.0

The `tox` stacktrace for completeness

```
GLOB sdist-make: /Users/johannes/code/reference-qvm/setup.py
py27 recreate: /Users/johannes/code/reference-qvm/.tox/py27
py27 installdeps: -rrequirements.txt
py27 inst: /Users/johannes/code/reference-qvm/.tox/dist/referenceqvm-0.1.2.zip
py27 installed: antlr4-python2-runtime==4.7,backports.functools-lru-cache==1.4,certifi==2017.11.5,chardet==3.0.4,codeclimate-test-reporter==0.2.3,coverage==4.4.2,cycler==0.10.0,funcsigs==1.0.2,idna==2.6,matplotlib==2.1.0,numpy==1.13.3,pluggy==0.5.2,py==1.5.2,pyparsing==2.2.0,pyquil==1.4.0,pytest==3.2.5,pytest-cov==2.5.1,python-dateutil==2.6.1,pytz==2017.3,referenceqvm==0.1.2,requests==2.18.4,scipy==1.0.0,six==1.11.0,sphinx-rtd-theme==0.2.4,subprocess32==3.2.7,tox==2.9.1,urllib3==1.22,virtualenv==15.1.0
py27 runtests: PYTHONHASHSEED='4114467311'
py27 runtests: commands[0] | py.test --cov=referenceqvm
====================================================================================================================================================================== test session starts ======================================================================================================================================================================
platform darwin -- Python 2.7.13, pytest-3.2.5, py-1.5.2, pluggy-0.4.0
rootdir: /Users/johannes/code/reference-qvm, inifile:
plugins: cov-2.5.1
collected 1 item / 8 errors                                                                                                                                                                                                                                                                                                                                      


---------- coverage: platform darwin, python 2.7.13-final-0 ----------
Name                                Stmts   Miss  Cover
-------------------------------------------------------
referenceqvm/__init__.py                1      0   100%
referenceqvm/api.py                    10     10     0%
referenceqvm/gates.py                  40     11    73%
referenceqvm/qam.py                    79     79     0%
referenceqvm/qvm_unitary.py            23     23     0%
referenceqvm/qvm_wavefunction.py      140    140     0%
referenceqvm/unitary_generator.py     129    122     5%
-------------------------------------------------------
TOTAL                                 422    385     9%

============================================================================================================================================================================ ERRORS =============================================================================================================================================================================
_______________________________________________________________________________________________________________________________________________ ERROR collecting referenceqvm/tests/test_arbitrary_state_prep.py ________________________________________________________________________________________________________________________________________________
ImportError while importing test module '/Users/johannes/code/reference-qvm/referenceqvm/tests/test_arbitrary_state_prep.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
referenceqvm/tests/test_arbitrary_state_prep.py:3: in <module>
    from pyquil.quil import Program
.tox/py27/lib/python2.7/site-packages/pyquil/quil.py:25: in <module>
    from pyquil._parser.PyQuilListener import run_parser
.tox/py27/lib/python2.7/site-packages/pyquil/_parser/PyQuilListener.py:18: in <module>
    from typing import Any, List
E   ImportError: No module named typing
____________________________________________________________________________________________________________________________________________________ ERROR collecting referenceqvm/tests/test_controlflow.py ____________________________________________________________________________________________________________________________________________________
ImportError while importing test module '/Users/johannes/code/reference-qvm/referenceqvm/tests/test_controlflow.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
referenceqvm/tests/test_controlflow.py:2: in <module>
    from pyquil.quil import Program
.tox/py27/lib/python2.7/site-packages/pyquil/quil.py:25: in <module>
    from pyquil._parser.PyQuilListener import run_parser
.tox/py27/lib/python2.7/site-packages/pyquil/_parser/PyQuilListener.py:18: in <module>
    from typing import Any, List
E   ImportError: No module named typing
____________________________________________________________________________________________________________________________________________________ ERROR collecting referenceqvm/tests/test_measurement.py ____________________________________________________________________________________________________________________________________________________
ImportError while importing test module '/Users/johannes/code/reference-qvm/referenceqvm/tests/test_measurement.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
referenceqvm/tests/test_measurement.py:2: in <module>
    from pyquil.quil import Program
.tox/py27/lib/python2.7/site-packages/pyquil/quil.py:25: in <module>
    from pyquil._parser.PyQuilListener import run_parser
.tox/py27/lib/python2.7/site-packages/pyquil/_parser/PyQuilListener.py:18: in <module>
    from typing import Any, List
E   ImportError: No module named typing
_____________________________________________________________________________________________________________________________________________________ ERROR collecting referenceqvm/tests/test_qvm_funcs.py _____________________________________________________________________________________________________________________________________________________
ImportError while importing test module '/Users/johannes/code/reference-qvm/referenceqvm/tests/test_qvm_funcs.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
referenceqvm/tests/test_qvm_funcs.py:1: in <module>
    from pyquil.quil import Program
.tox/py27/lib/python2.7/site-packages/pyquil/quil.py:25: in <module>
    from pyquil._parser.PyQuilListener import run_parser
.tox/py27/lib/python2.7/site-packages/pyquil/_parser/PyQuilListener.py:18: in <module>
    from typing import Any, List
E   ImportError: No module named typing
______________________________________________________________________________________________________________________________________________________ ERROR collecting referenceqvm/tests/test_system.py _______________________________________________________________________________________________________________________________________________________
ImportError while importing test module '/Users/johannes/code/reference-qvm/referenceqvm/tests/test_system.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
referenceqvm/tests/test_system.py:1: in <module>
    from pyquil.quil import Program
.tox/py27/lib/python2.7/site-packages/pyquil/quil.py:25: in <module>
    from pyquil._parser.PyQuilListener import run_parser
.tox/py27/lib/python2.7/site-packages/pyquil/_parser/PyQuilListener.py:18: in <module>
    from typing import Any, List
E   ImportError: No module named typing
______________________________________________________________________________________________________________________________________________________ ERROR collecting referenceqvm/tests/test_unitary.py ______________________________________________________________________________________________________________________________________________________
ImportError while importing test module '/Users/johannes/code/reference-qvm/referenceqvm/tests/test_unitary.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
referenceqvm/tests/test_unitary.py:1: in <module>
    from pyquil.quil import Program
.tox/py27/lib/python2.7/site-packages/pyquil/quil.py:25: in <module>
    from pyquil._parser.PyQuilListener import run_parser
.tox/py27/lib/python2.7/site-packages/pyquil/_parser/PyQuilListener.py:18: in <module>
    from typing import Any, List
E   ImportError: No module named typing
_________________________________________________________________________________________________________________________________________________ ERROR collecting referenceqvm/tests/test_unitary_generator.py _________________________________________________________________________________________________________________________________________________
ImportError while importing test module '/Users/johannes/code/reference-qvm/referenceqvm/tests/test_unitary_generator.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
referenceqvm/tests/test_unitary_generator.py:4: in <module>
    from referenceqvm.unitary_generator import (lifted_gate, apply_gate,
referenceqvm/unitary_generator.py:30: in <module>
    from pyquil.paulis import PauliSum
.tox/py27/lib/python2.7/site-packages/pyquil/paulis.py:24: in <module>
    from .quil import Program
.tox/py27/lib/python2.7/site-packages/pyquil/quil.py:25: in <module>
    from pyquil._parser.PyQuilListener import run_parser
.tox/py27/lib/python2.7/site-packages/pyquil/_parser/PyQuilListener.py:18: in <module>
    from typing import Any, List
E   ImportError: No module named typing
___________________________________________________________________________________________________________________________________________________ ERROR collecting referenceqvm/tests/test_wavefunction.py ____________________________________________________________________________________________________________________________________________________
ImportError while importing test module '/Users/johannes/code/reference-qvm/referenceqvm/tests/test_wavefunction.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
referenceqvm/tests/test_wavefunction.py:4: in <module>
    from pyquil.quil import Program
.tox/py27/lib/python2.7/site-packages/pyquil/quil.py:25: in <module>
    from pyquil._parser.PyQuilListener import run_parser
.tox/py27/lib/python2.7/site-packages/pyquil/_parser/PyQuilListener.py:18: in <module>
    from typing import Any, List
E   ImportError: No module named typing
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 8 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
==================================================================================================================================================================== 8 error in 0.86 seconds ====================================================================================================================================================================
ERROR: InvocationError: '/Users/johannes/code/reference-qvm/.tox/py27/bin/py.test --cov=referenceqvm'
py36 recreate: /Users/johannes/code/reference-qvm/.tox/py36
py36 installdeps: -rrequirements.txt
py36 inst: /Users/johannes/code/reference-qvm/.tox/dist/referenceqvm-0.1.2.zip
py36 installed: antlr4-python3-runtime==4.7,certifi==2017.11.5,chardet==3.0.4,codeclimate-test-reporter==0.2.3,coverage==4.4.2,cycler==0.10.0,funcsigs==1.0.2,idna==2.6,matplotlib==2.1.0,numpy==1.13.3,pluggy==0.5.2,py==1.5.2,pyparsing==2.2.0,pyquil==1.4.0,pytest==3.2.5,pytest-cov==2.5.1,python-dateutil==2.6.1,pytz==2017.3,referenceqvm==0.1.2,requests==2.18.4,scipy==1.0.0,six==1.11.0,sphinx-rtd-theme==0.2.4,tox==2.9.1,urllib3==1.22,virtualenv==15.1.0
py36 runtests: PYTHONHASHSEED='4114467311'
py36 runtests: commands[0] | py.test --cov=referenceqvm
====================================================================================================================================================================== test session starts ======================================================================================================================================================================
platform darwin -- Python 3.6.1, pytest-3.2.5, py-1.5.2, pluggy-0.4.0
rootdir: /Users/johannes/code/reference-qvm, inifile:
plugins: cov-2.5.1
collected 22 items / 1 errors                                                                                                                                                                                                                                                                                                                                    


---------- coverage: platform darwin, python 3.6.1-final-0 -----------
Name                                Stmts   Miss  Cover
-------------------------------------------------------
referenceqvm/__init__.py                1      0   100%
referenceqvm/api.py                    10     10     0%
referenceqvm/gates.py                  40     11    72%
referenceqvm/qam.py                    79     79     0%
referenceqvm/qvm_unitary.py            23     23     0%
referenceqvm/qvm_wavefunction.py      140    135     4%
referenceqvm/unitary_generator.py     129    111    14%
-------------------------------------------------------
TOTAL                                 422    369    13%

============================================================================================================================================================================ ERRORS =============================================================================================================================================================================
___________________________________________________________________________________________________________________________________________________ ERROR collecting referenceqvm/tests/test_wavefunction.py ____________________________________________________________________________________________________________________________________________________
ImportError while importing test module '/Users/johannes/code/reference-qvm/referenceqvm/tests/test_wavefunction.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
referenceqvm/tests/test_wavefunction.py:15: in <module>
    from referenceqvm.qvm_wavefunction import QVM_Wavefunction
referenceqvm/qvm_wavefunction.py:45: in <module>
    from pyquil.quilbase import (Instr,
E   ImportError: cannot import name 'Instr'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
==================================================================================================================================================================== 1 error in 0.58 seconds ====================================================================================================================================================================
ERROR: InvocationError: '/Users/johannes/code/reference-qvm/.tox/py36/bin/py.test --cov=referenceqvm'
____________________________________________________________________________________________________________________________________________________________________________ summary ____________________________________________________________________________________________________________________________________________________________________________
ERROR:   py27: commands failed
ERROR:   py36: commands failed
```

--- a/referenceqvm/qam.py
+++ b/referenceqvm/qam.py
@@ -23,7 +23,7 @@ import numpy as np
 from referenceqvm.unitary_generator import value_get
 
 from pyquil.quil import Program
-from pyquil.quilbase import (Instr,
+from pyquil.quilbase import (Gate,
                              Measurement,
                              UnaryClassicalInstruction,
                              BinaryClassicalInstruction)
@@ -72,9 +72,6 @@ class QAM(object):
             raise NotImplementedError("QAM needs to be subclassed in order to "
                                       "load program")
 
-        # synthesize program into instruction list
-        synthesized_prog = pyquil_program.synthesize()
-
         # create defgate dictionary
         defined_gates = {}
         for dg in pyquil_program.defined_gates:
@@ -83,10 +80,9 @@ class QAM(object):
 
         # if QVM_Unitary, check if all instructions are valid.
         invalid = False
-        for index, action in enumerate(synthesized_prog):
-            if isinstance(action, Instr):
-                if (action.operator_name not in self.gate_set.keys()
-                   and action.operator_name not in self.defgate_set.keys()):
+        for instr in pyquil_program:
+            if isinstance(instr, Gate):
+                if not (instr.name in self.gate_set.keys() or instr.name in self.defgate_set.keys()):
                     invalid = True
                     break
             else:
@@ -98,7 +94,7 @@ class QAM(object):
                             "supported")
 
         # set internal program and counter to their appropriate values
-        self.program = synthesized_prog
+        self.program = pyquil_program
         self.program_counter = 0
 
         # setup quantum and classical memory
@@ -121,10 +117,10 @@ class QAM(object):
         for index, inst in enumerate(self.program):
             if isinstance(inst, Measurement):
                 # instruction is measurement, acts on qbits and cbits
-                if value_get(inst.arguments[0]) > q_max:
-                    q_max = value_get(inst.arguments[0])
-                elif value_get(inst.arguments[1]) > c_max:
-                    c_max = value_get(inst.arguments[1])
+                if value_get(inst.qubit) > q_max:
+                    q_max = value_get(inst.qubit)
+                elif value_get(inst.classical_reg) > c_max:
+                    c_max = value_get(inst.classical_reg)
             elif isinstance(inst, UnaryClassicalInstruction):
                 # instruction acts on cbits
                 if value_get(inst.target) > c_max:
@@ -135,10 +131,9 @@ class QAM(object):
                     c_max = value_get(inst.left)
                 elif value_get(inst.right) > c_max:
                     c_max = value_get(inst.right)
-            elif isinstance(inst, Instr):
-                # instruction is Gate or DefGate, acts on qbits
-                if max(map(lambda x: value_get(x), inst.arguments)) > q_max:
-                    q_max = max(map(lambda x: value_get(x), inst.arguments))
+            elif isinstance(inst, Gate):
+                if max(map(lambda x: value_get(x), inst.qubits)) > q_max:
+                    q_max = max(map(lambda x: value_get(x), inst.qubits))
         q_max += 1  # 0-indexed
         c_max += 1  # 0-indexed
         q_limit = 51

--- a/referenceqvm/qvm_unitary.py
+++ b/referenceqvm/qvm_unitary.py
@@ -59,10 +59,9 @@ class QVM_Unitary(QAM):
         """
         Implements a transition on the unitary-qvm.
 
-        :param instruction: QuilAction gate to be implemented
+        :param Gate instruction: QuilAction gate to be implemented
         """
-        if instruction.operator_name in self.gate_set or \
-            instruction.operator_name in self.defgate_set:
+        if instruction.name in self.gate_set or instruction.name in self.defgate_set:
             # get the unitary and evolve the state
             unitary = tensor_gates(self.gate_set, self.defgate_set,
                                    instruction, self.num_qubits)
@@ -70,7 +69,7 @@ class QVM_Unitary(QAM):
             self.program_counter += 1
         else:
             raise TypeError("Gate {} is not in the "
-                            "gate set".format(instruction.operator_name))
+                            "gate set".format(instruction.name))
 
     def unitary(self, pyquil_program):
         """

--- a/referenceqvm/qvm_wavefunction.py
+++ b/referenceqvm/qvm_wavefunction.py
@@ -42,8 +42,7 @@ import numpy as np
 import scipy.sparse as sps
 
 from pyquil.quil import Program
-from pyquil.quilbase import (Instr,
-                             Measurement,
+from pyquil.quilbase import (Measurement,
                              UnaryClassicalInstruction,
                              BinaryClassicalInstruction,
                              ClassicalTrue,
@@ -59,7 +58,7 @@ from pyquil.quilbase import (Instr,
                              JumpConditional,
                              JumpWhen,
                              JumpUnless,
-                             Halt)
+                             Halt, Gate)
 from pyquil.wavefunction import Wavefunction
 from referenceqvm.unitary_generator import lifted_gate, tensor_gates, value_get
 from referenceqvm.gates import utility_gates
@@ -167,8 +166,8 @@ class QVM_Wavefunction(QAM):
         """
         if isinstance(instruction, Measurement):
             # perform measurement and modify wf in-place
-            t_qbit = value_get(instruction.arguments[0])
-            t_cbit = value_get(instruction.arguments[1])
+            t_qbit = value_get(instruction.qubit)
+            t_cbit = value_get(instruction.classical_reg)
             measured_val, unitary = self.measurement(t_qbit, psi=None)
             self.wf = unitary.dot(self.wf)
 
@@ -176,7 +175,7 @@ class QVM_Wavefunction(QAM):
             self.classical_memory[t_cbit] = measured_val
             self.program_counter += 1
 
-        elif isinstance(instruction, Instr):
+        elif isinstance(instruction, Gate):
             # apply Gate or DefGate
             unitary = tensor_gates(self.gate_set, self.defgate_set, instruction, self.num_qubits)
             self.wf = unitary.dot(self.wf)

--- a/referenceqvm/tests/test_unitary_generator.py
+++ b/referenceqvm/tests/test_unitary_generator.py
@@ -176,39 +176,39 @@ def test_single_qubit_gates():
 
 def test_tensor_gates_single_qubit():
     prog = Program().inst([Hgate(0)])
-    test_unitary = tensor_gates(gate_matrix, {}, prog.actions[0][1], 1).toarray()
+    test_unitary = tensor_gates(gate_matrix, {}, prog.instructions[0], 1).toarray()
     true_unitary = gate_matrix['H']
     assert np.allclose(test_unitary, true_unitary)
 
     prog = Program().inst([Hgate(0)])
-    test_unitary = tensor_gates(gate_matrix, {}, prog.actions[0][1], 5).toarray()
+    test_unitary = tensor_gates(gate_matrix, {}, prog.instructions[0], 5).toarray()
     true_unitary = np.kron(np.eye(2**4), gate_matrix['H'])
     assert np.allclose(test_unitary, true_unitary)
 
     prog = Program().inst([RXgate(0.2)(3)])
-    test_unitary = tensor_gates(gate_matrix, {}, prog.actions[0][1], 5).toarray()
+    test_unitary = tensor_gates(gate_matrix, {}, prog.instructions[0], 5).toarray()
     true_unitary = np.kron(np.eye(2**1), np.kron(gate_matrix['RX'](0.2),  np.eye(2**3)))
     assert np.allclose(test_unitary, true_unitary)
 
     prog = Program().inst([RXgate(0.5)(4)])
-    test_unitary = tensor_gates(gate_matrix, {}, prog.actions[0][1], 5).toarray()
+    test_unitary = tensor_gates(gate_matrix, {}, prog.instructions[0], 5).toarray()
     true_unitary = np.kron(np.eye(2**0), np.kron(gate_matrix['RX'](0.5),  np.eye(2**4)))
     assert np.allclose(test_unitary, true_unitary)
 
 
 def test_tensor_gates_two_qubit():
     prog = Program().inst([CNOTgate(0, 1)])
-    test_unitary = tensor_gates(gate_matrix, {}, prog.actions[0][1], 4).toarray()
+    test_unitary = tensor_gates(gate_matrix, {}, prog.instructions[0], 4).toarray()
     true_unitary = apply_gate(gate_matrix['CNOT'], [0, 1], 4).toarray()
     assert np.allclose(test_unitary, true_unitary)
 
     prog = Program().inst([CNOTgate(1, 0)])
-    test_unitary = tensor_gates(gate_matrix, {}, prog.actions[0][1], 4).toarray()
+    test_unitary = tensor_gates(gate_matrix, {}, prog.instructions[0], 4).toarray()
     true_unitary = apply_gate(gate_matrix['CNOT'], [1, 0], 4).toarray()
     assert np.allclose(test_unitary, true_unitary)
 
     prog = Program().inst([CNOTgate(1, 3)])
-    test_unitary = tensor_gates(gate_matrix, {}, prog.actions[0][1], 4).toarray()
+    test_unitary = tensor_gates(gate_matrix, {}, prog.instructions[0], 4).toarray()
     true_unitary = apply_gate(gate_matrix['CNOT'], [1, 3], 4).toarray()
     assert np.allclose(test_unitary, true_unitary)
 

--- a/referenceqvm/unitary_generator.py
+++ b/referenceqvm/unitary_generator.py
@@ -23,7 +23,6 @@ timeefficiency.
 """
 from collections import Sequence
 from numbers import Integral
-import warnings
 
 import scipy.sparse as sps
 from pyquil.quilbase import *
@@ -327,16 +326,16 @@ def tensor_gates(gate_set, defgate_set, pyquil_gate, num_qubits):
 
     :param dict gate_set: gate dictionary (name, matrix) pairs
     :param dict defgate_set: defined gate dictionary (name, matrix) pairs
-    :param Instr pyquil_gate: Instruction object for pyQuil gate
+    :param Gate pyquil_gate: Instruction object for pyQuil gate
     :param int num_qubits: number of qubits in Hilbert space
 
     :return: input gate lifted to full Hilbert space and applied
     :rtype: np.array
     """
-    if pyquil_gate.operator_name in gate_set:
+    if pyquil_gate.name in gate_set:
         # Input gate set. Assumed to be standard gate set.
         dict_check = gate_set
-    elif pyquil_gate.operator_name in defgate_set:
+    elif pyquil_gate.name in defgate_set:
         # defined_gates
         dict_check = defgate_set
     else:
@@ -344,16 +343,16 @@ def tensor_gates(gate_set, defgate_set, pyquil_gate, num_qubits):
                          "found in standard gate set or defined "
                          "gate set of program!")
 
-    args = tuple(value_get(x) for x in pyquil_gate.arguments) \
-            if dict_check == gate_matrix else tuple(pyquil_gate.arguments)
+    args = tuple(value_get(x) for x in pyquil_gate.qubits) \
+            if dict_check == gate_matrix else tuple(pyquil_gate.qubits)
 
-    if pyquil_gate.parameters:
-        gate = apply_gate(dict_check[pyquil_gate.operator_name]
-                          (*[value_get(p) for p in pyquil_gate.parameters]),
+    if pyquil_gate.params:
+        gate = apply_gate(dict_check[pyquil_gate.name]
+                          (*[value_get(p) for p in pyquil_gate.params]),
                           args,
                           num_qubits)
     else:
-        gate = apply_gate(dict_check[pyquil_gate.operator_name],
+        gate = apply_gate(dict_check[pyquil_gate.name],
                           args,
                           num_qubits)
 
@@ -412,8 +411,8 @@ def value_get(param_obj):
     """
     if isinstance(param_obj, (float, int)):
         return param_obj
-    elif isinstance(param_obj, AbstractQubit):
-        return param_obj.index()
+    elif isinstance(param_obj, Qubit):
+        return param_obj.index
     elif isinstance(param_obj, Addr):
         return param_obj.address
     elif isinstance(param_obj, Slot):

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,6 +1,6 @@
 numpy>=1.11.1
 scipy>=0.18.1
-pyquil==1.3.0
+pyquil==1.4.1
 funcsigs
 
 # For testing

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,6 +1,6 @@
 numpy>=1.11.1
 scipy>=0.18.1
-pyquil==1.4.1
+pyquil==1.4.2
 funcsigs
 
 # For testing

--- a/setup.py
+++ b/setup.py
@@ -16,7 +16,7 @@ setup(
     install_requires=[
         "numpy >= 1.11.1",
         "scipy >= 0.18.1",
-        "pyquil >= 1.4.1",
+        "pyquil >= 1.4.2",
         "funcsigs"
     ],
     setup_requires=['pytest-runner'],

--- a/setup.py
+++ b/setup.py
@@ -16,7 +16,7 @@ setup(
     install_requires=[
         "numpy >= 1.11.1",
         "scipy >= 0.18.1",
-        "pyquil >= 1.1.1",
+        "pyquil >= 1.4.1",
         "funcsigs"
     ],
     setup_requires=['pytest-runner'],

